File: Java/src/com/bh/no0058_Length_of_Last_Word.java
package com.bh;

public class no0058_Length_of_Last_Word {

    static class Solution {
        public int lengthOfLastWord(String s) {
            // Start from the end of the string
            int i = s.length() - 1;

            // Skip trailing spaces
            while (i >= 0 && s.charAt(i) == ' ') {
                i--;
            }

            // Count the length of the last word
            int length = 0;
            while (i >= 0 && s.charAt(i) != ' ') {
                length++;
                i--;
            }

            return length;
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // Test case 1
        String test1 = "Hello World";
        System.out.println("Input: \"" + test1 + "\"");
        System.out.println("Output: " + solution.lengthOfLastWord(test1));
        System.out.println("Expected: 5\n");

        // Test case 2
        String test2 = "   fly me   to   the moon  ";
        System.out.println("Input: \"" + test2 + "\"");
        System.out.println("Output: " + solution.lengthOfLastWord(test2));
        System.out.println("Expected: 4\n");

        // Test case 3
        String test3 = "luffy is still joyboy";
        System.out.println("Input: \"" + test3 + "\"");
        System.out.println("Output: " + solution.lengthOfLastWord(test3));
        System.out.println("Expected: 6\n");

        // Additional test cases
        String test4 = "a";
        System.out.println("Input: \"" + test4 + "\"");
        System.out.println("Output: " + solution.lengthOfLastWord(test4));
        System.out.println("Expected: 1\n");

        String test5 = " a ";
        System.out.println("Input: \"" + test5 + "\"");
        System.out.println("Output: " + solution.lengthOfLastWord(test5));
        System.out.println("Expected: 1");
    }
}

File: CPlusPlus/no0058_Length_of_Last_Word.cpp
//
// Created by binghuan on 2025/9/7.
//

#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        // Start from the end and skip trailing spaces
        int end = s.length() - 1;
        while (end >= 0 && s[end] == ' ') {
            end--;
        }

        // Count the length of the last word
        int length = 0;
        while (end >= 0 && s[end] != ' ') {
            length++;
            end--;
        }

        return length;
    }
};

int main() {
    Solution solution;

    // Test cases
    cout << "Test Case 1:" << endl;
    string s1 = "Hello World";
    cout << "Input: \"" << s1 << "\"" << endl;
    cout << "Output: " << solution.lengthOfLastWord(s1) << endl;
    cout << "Expected: 5" << endl << endl;

    cout << "Test Case 2:" << endl;
    string s2 = "   fly me   to   the moon  ";
    cout << "Input: \"" << s2 << "\"" << endl;
    cout << "Output: " << solution.lengthOfLastWord(s2) << endl;
    cout << "Expected: 4" << endl << endl;

    cout << "Test Case 3:" << endl;
    string s3 = "luffy is still joyboy";
    cout << "Input: \"" << s3 << "\"" << endl;
    cout << "Output: " << solution.lengthOfLastWord(s3) << endl;
    cout << "Expected: 6" << endl << endl;

    // Additional edge cases
    cout << "Test Case 4 (single word):" << endl;
    string s4 = "hello";
    cout << "Input: \"" << s4 << "\"" << endl;
    cout << "Output: " << solution.lengthOfLastWord(s4) << endl;
    cout << "Expected: 5" << endl << endl;

    cout << "Test Case 5 (word with leading spaces):" << endl;
    string s5 = "   hello";
    cout << "Input: \"" << s5 << "\"" << endl;
    cout << "Output: " << solution.lengthOfLastWord(s5) << endl;
    cout << "Expected: 5" << endl << endl;

    return 0;
}
